### PR TITLE
Add extension to provide custom builder image environment variables

### PR DESCRIPTION
**Pull request by rottencandy**
**Story**: 
https://issues.redhat.com/browse/ODC-6379

**Description**: 
Create an extension that allows specifying extra build environment options for builder images in git import form.
Use the extension to provide option to customize npm run command(`NPM_RUN`) for nodejs builder images.

**Screen shots / Gifs for design review**: 

**Unit test coverage report**: 
(Unchanged)

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by jerolimov** [comment]
Since the epic acceptance criteria have not changed and the epic ODC-6266 has a doc-ack and px-ack, I approve this on behalf of doc and PX team.

/label docs-approved
/label px-approved

File: frontend/packages/dev-console/src/components/import/import-submit-utils.ts
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/import/import-submit-utils.ts]:
```suggestion
  const customBuildEnvs = imageEnv
    ? Object.keys(imageEnv).map((k) => ({ name: k, value: imageEnv[k] }))
    : [];
```

As we noticed together in slack, this code should check if the value is not empty.

Otherwise a touch to the `NPM_RUN` field would broke the deployment. (Builld passes, but the app could not started with `NPM_RUN=` environment variable.

File: frontend/packages/console-dynamic-plugin-sdk/src/extensions/import-environments.ts
Comment (jerolimov) [inline frontend/packages/console-dynamic-plugin-sdk/src/extensions/import-environments.ts:7]: Can we add an optional description here? This would allow us to describe where the "Run command" is saved.

File: frontend/packages/dev-console/console-extensions.json
Comment (jerolimov) [inline frontend/packages/dev-console/console-extensions.json:197]: Same as above, a description would be helpful, or?

```suggestion
          "label": "%devconsole~Run command%",
          "description": "%devconsole~Optional parameter for Node.js applications to define the npm run <command>. It is saved as environment variable NPM_RUN of the BuildConfig.%",
```

File: frontend/packages/dev-console/src/components/import/builder/BuilderImageEnvironments.tsx
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/import/builder/BuilderImageEnvironments.tsx:48]:
```suggestion
            label={t(env.label)}
            helpText={env.description ? t(env.description) : null}
```

**Review by jerolimov — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/src/components/import/import-submit-utils.ts
Comment (jerolimov) [reply frontend/packages/dev-console/src/components/import/import-submit-utils.ts]: Already solved :+1:

File: frontend/packages/dev-console/console-extensions.json
Comment (rottencandy) [reply frontend/packages/dev-console/console-extensions.json:197]: Updated.

**Comment by jerolimov** [comment]
Thanks for adopting the suggestions. Retested this now on a cluster and touching the field doesn't break the Deployment anymore. :+1: 

UI looks  fine for me, but we should now wait for a lgtm from UX/PM. /cc @serenamarie125 @beaumorley 

/label qe-approved
/approve

**Review by jerolimov — APPROVED** [review]

File: frontend/packages/dev-console/src/components/import/builder/BuilderImageEnvironments.tsx
Comment (sahil143) [inline frontend/packages/dev-console/src/components/import/builder/BuilderImageEnvironments.tsx:49]: I don't think translate is needed here for env.description it's done when extension is defined

Comment (rottencandy) [reply frontend/packages/dev-console/src/components/import/builder/BuilderImageEnvironments.tsx]: Updated.

**Comment by beaumorley** [comment]
Wondering if we need all the help text under the form field? Could we remove the second sentence? 

image

**Comment by rottencandy** [comment]
Updated:

**Comment by jerolimov** [comment]
The 2nd sentence explain something which the user may not know. Alternative was a question mark icon..

But for now I'm fine with the shortened version as well. It's look good for me. So with this change this is fine for you/UX aswell, right?

I will the just run the e2e tests tomorrow and if this will pass I would like to add lgtm to get this merged :)

File: frontend/packages/console-dynamic-plugin-sdk/src/extensions/import-environments.ts
Comment (christianvogt) [inline frontend/packages/console-dynamic-plugin-sdk/src/extensions/import-environments.ts:9]: My first thought was this should be named `value` or `defaultValue`.

Comment (rottencandy) [reply frontend/packages/console-dynamic-plugin-sdk/src/extensions/import-environments.ts]: Updated to `defaultValue`.

**Comment by jerolimov** [comment]
Tested this again on a cluster and works fine. Christian comment is implemented, so adding finally lgtm.

/lgtm

File: frontend/packages/dev-console/console-extensions.json
Comment (jerolimov) [inline frontend/packages/dev-console/console-extensions.json:193]: @rottencandy @christianvogt Any thoughts here if we should also add node `16-` or `ubi9` here already or should we update this shortly before we release 4.10? ... I will create a story so that we can check in 2-3 month which versions we should support.

See https://github.com/sclorg/s2i-nodejs-container/pull/295

**Comment by jerolimov** [comment]
Stop rerunning tests until we take a new look why this fails.
/hold

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-git.feature
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/features/addFlow/create-from-git.feature:207]: Btw, it would be still cool if this is not a manual (manually clicked through the UI) test and the statements below could run via cypress.

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-git.feature
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/features/addFlow/create-from-git.feature:211]: That is not what the user does. The user enters a "Run command", not a build environment variable.

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-git.feature
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/features/addFlow/create-from-git.feature:215]: I'm sure the user needs to click on "Build #1" first, switch then to the environment variable tab and then can verify that the NPM_RUN variable is set.

File: frontend/packages/dev-console/console-extensions.json
Comment (rottencandy) [reply frontend/packages/dev-console/console-extensions.json]: Don't think it would affect anything so added `16-ubi8` to the list.

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-git.feature
Comment (rottencandy) [reply frontend/packages/dev-console/integration-tests/features/addFlow/create-from-git.feature]: Yup, removed manual. Since it is only a single API call, I don't think it would affect the rate limit issue that much.

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-git.feature
Comment (rottencandy) [reply frontend/packages/dev-console/integration-tests/features/addFlow/create-from-git.feature]: Updated.

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-git.feature
Comment (rottencandy) [reply frontend/packages/dev-console/integration-tests/features/addFlow/create-from-git.feature]: Updated to click on build instead of BuildConfig and changed statement text to reflect it. Or were you suggesting we do this in multiple separate steps instead of one?

**Comment by jerolimov** [comment]
Thanks @rottencandy, looks much better. My local cypress has some (unrelated) issues. Let check if the tests pass on CI, but the code is fine :+1: 

/lgtm
/unhold

**Pull request merged**
